frontend/lib/resolution/resolution-error-classes-list.cpp

```diff
@@ -840,7 +840,13 @@ void ErrorNoMatchingCandidates::write(ErrorWriterBase& wr) const {
       wr.note(fn->id(), "the following candidate didn't match because an actual couldn't be passed to a formal:");
       wr.code(fn->id(), { formalDecl });
 
-      wr.message("The formal '", formalDecl->name(), "' expects ", badPass.formalType(), ", but the actual was ", badPass.actualType(), ".");
+      std::string formalName;
+      if (auto named = formalDecl->toNamedDecl()) {
+        formalName = "'" + named->name().str() + "'";
+      } else if (formalDecl->isTupleDecl()) {
+        // TODO: should we do anything fancy here?
+      }
```

Comment:
We should probably have an `else` that crashes if we think it's unreachable, or otherwise something different to print if we don't know the formal name